main/src/library/LazyList.flix

```diff
@@ -181,4 +195,61 @@ namespace LazyList {
         case LCons(x, xs) => mapEAcc(f, force xs, ECons(f(x), acc))
         case LList(xs)    => mapEAcc(f, force xs, acc)
     }
+
+    ///
+    /// Applies `f` to a start value `s` and all elements in `l` going from left to right.
+    ///
+    /// That is, the result is of the form: `f(...f(f(s, x1), x2)..., xn)`.
+    ///
+    pub def foldLeft(f: (b, a) -> b & ef, s: b, l: LazyList[a]): b & ef = match l {
+        case ENil         => s
+        case ECons(x, xs) => foldLeft(f, f(s, x),       xs)
+        case LCons(x, xs) => foldLeft(f, f(s, x), force xs)
+        case LList(xs)    => foldLeft(f, s,       force xs)
+    }
+
+    ///
+    /// Returns `k` appended to `l`.
+    ///
+    pub def append(l: LazyList[a], k: LazyList[a]): LazyList[a] = match l {
```

Comment:
While our naming is very inconsistent:

![image](https://user-images.githubusercontent.com/2287953/130347944-a7d5cbbe-f55f-4044-8738-92e0998174f2.png)

I think the right naming here should be `l1` and `l2` over `l` and `k`.